signer/tests/integration/postgres.rs

```diff
@@ -1867,3 +1868,474 @@ async fn can_store_and_get_latest_sweep_transaction() {
+    let test_data = TestData::generate(&mut rng, &signer_set, &test_params);
+    test_data.write_to(&db).await;
+
+    // Let's write the deposit request and associated trasnaction to the
+    // database. Here the deposit transaction will be confirmed on the
+    // canonical bitcoin blockchain.
+    let deposit_request: model::DepositRequest = fake::Faker.fake_with_rng(&mut rng);
+    let chain_tip = db.get_bitcoin_canonical_chain_tip().await.unwrap().unwrap();
+    let txid = &deposit_request.txid;
+    let output_index = deposit_request.output_index;
+    let signer_public_key = &signer_set[0];
+
+    let tx = model::Transaction {
+        txid: deposit_request.txid.into_bytes(),
+        tx: Vec::new(),
+        tx_type: model::TransactionType::DepositRequest,
+        block_hash: chain_tip.into_bytes(),
+    };
+    let tx_ref = model::BitcoinTxRef {
+        txid: deposit_request.txid,
+        block_hash: chain_tip,
+    };
+
+    db.write_deposit_request(&deposit_request).await.unwrap();
+    db.write_transaction(&tx).await.unwrap();
+    db.write_bitcoin_transaction(&tx_ref).await.unwrap();
+
+    // Write the decision to the database
+    let mut decision: model::DepositSigner = fake::Faker.fake_with_rng(&mut rng);
+    decision.output_index = deposit_request.output_index;
+    decision.txid = deposit_request.txid;
+    decision.signer_pub_key = *signer_public_key;
+
+    db.write_deposit_signer_decision(&decision).await.unwrap();
+
+    // Okay now let's pretend that the deposit has been swept, but the
+    // sweep gets reorged. For that we need a row in the `sweep_*` tables,
+    // and records in the `transactions` and `bitcoin_transactions` tables,
+    // but we'll use a random block for what confirms the sweep transaction
+    // so that it is not on the canonical bitcoin blockchain.
+    let random_block: model::BitcoinBlock = fake::Faker.fake_with_rng(&mut rng);
+    let mut sweep_tx: model::SweepTransaction = fake::Faker.fake_with_rng(&mut rng);
+    sweep_tx.created_at_block_hash = chain_tip;
+    sweep_tx.swept_withdrawals = Vec::new();
+    sweep_tx.swept_deposits = vec![model::SweptDeposit {
+        input_index: 1,
+        deposit_request_output_index: deposit_request.output_index,
+        deposit_request_txid: deposit_request.txid,
+    }];
+
+    let sweep_tx_model = model::Transaction {
+        tx_type: model::TransactionType::SbtcTransaction,
+        txid: sweep_tx.txid.to_byte_array(),
+        tx: Vec::new(),
+        block_hash: random_block.block_hash.to_byte_array(),
+    };
+    let sweep_tx_ref = model::BitcoinTxRef {
+        txid: sweep_tx.txid,
+        block_hash: random_block.block_hash,
+    };
+    db.write_bitcoin_block(&random_block).await.unwrap();
+    db.write_transaction(&sweep_tx_model).await.unwrap();
+    db.write_bitcoin_transaction(&sweep_tx_ref).await.unwrap();
+    db.write_sweep_transaction(&sweep_tx).await.unwrap();
+
+    let report = db
```

Comment:
you could probably add another branch to this that asserts that if you get the report using the fork's tip (which would be the case if that fork became the new canonical chain) that it returns the correct information

Reply:
Good call, done in https://github.com/stacks-network/sbtc/pull/749/commits/1b8c4bb3e1d63b83a4ab392957a2c07ed75cc4be.

Reply:
Nice!